insights/parsers/smartctl.py

```diff
@@ -35,12 +35,13 @@ class SMARTctl(Parser):
 
     def __init__(self, context):
         super(SMARTctl, self).__init__(context)
+        self.data = {}
         filename_re = re.compile(r'smartctl_-a_\.dev\.(?P<device>\w+)$')
         match = filename_re.search(context.path)
-        self.data = {}
         if match:
             self.device = match.group('device')
-        # Else warn?  fail?
+        else:
+            self.device = ''
```

Comment:
@PaulWay It seems like a failure of this match would indicate a bug in either filling in the `context`s `path` or a bug in the associated `CommandSpec` or the referenced `pre-command`.  As such, it seems that if the path doesn't match, we have a case where we probably should not trust data from the mapper at all.  If that's all true, then I think we should raise an `Error`.